Common/include/option_structure.hpp

```diff
@@ -1101,6 +1101,58 @@ inline SST_ParsedOptions ParseSSTOptions(const SST_OPTIONS *SST_Options, unsigne
   return SSTParsedOptions;
 }
 
+/*!
+ * \brief SST rough-wall boundary conditions Options
+ */
+enum class ROUGHSST_MODEL {
+  NONE,                 /*!< \brief No option / default no surface roughness applied. */
+  WILCOX1998,           /*!< \brief Wilcox 1998 boundary conditions for rough walls. */
+  WILCOX2006,           /*!< \brief Wilcox 2006 boundary conditions for rough walls / default version if roughness is applied. */
+  LIMITER_KNOPP,        /*!< \brief Knopp eddy viscosity limiter. */
+  LIMITER_AUPOIX,       /*!< \brief Aupoix eddy viscosity limiter. */
+};
+static const MapType<std::string, ROUGHSST_MODEL> RoughSST_Model_Map = {
+  MakePair("NONE", ROUGHSST_MODEL::NONE)
+  MakePair("WILCOX1998", ROUGHSST_MODEL::WILCOX1998)
+  MakePair("WILCOX2006", ROUGHSST_MODEL::WILCOX2006)
+  MakePair("LIMITER_KNOPP", ROUGHSST_MODEL::LIMITER_KNOPP)
+  MakePair("LIMITER_AUPOIX", ROUGHSST_MODEL::LIMITER_AUPOIX)
+};
+
+/*!
+ * \brief Structure containing parsed SST rough-wall boundary conditions options.
+ */
+struct ROUGH_SST_ParsedOptions {
+  ROUGHSST_MODEL version = ROUGHSST_MODEL::NONE;         /*!< \brief KWBC base model. */
+  bool wilcox1998 = false;                           /*!< \brief Use Wilcox boundary conditions for rough walls (1998). */
+  bool wilcox2006 = false;                          /*!< \brief Use Wilcox boundary conditions for rough walls (2006). */
+  bool limiter_knopp = false;                        /*!< \brief Use Knopp eddy viscosity limiter. */
+  bool limiter_aupoix = false;                      /*!< \brief Use Aupoix eddy viscosity limiter. */
+};
+
+/*!
+ * \brief Function to parse SST rough-wall boundary conditions options.
+ * \param[in] ROUGHSST_Options - Selected SST rough-wall boundary conditions option from config.
+ * \param[in] nROUGHSST_Options - Number of options selected.
+ * \param[in] rank - MPI rank.
+ * \return Struct with SST options.
+ */
+inline ROUGH_SST_ParsedOptions ParseROUGHSSTOptions(ROUGHSST_MODEL sstbcs_option) {
+    ROUGH_SST_ParsedOptions opts;
+    opts.version = sstbcs_option;
+
+    switch(sstbcs_option) {
+        case ROUGHSST_MODEL::WILCOX1998: opts.wilcox1998 = true; break;
+        case ROUGHSST_MODEL::WILCOX2006: opts.wilcox2006 = true; break;
+        case ROUGHSST_MODEL::LIMITER_KNOPP: opts.limiter_knopp = true; break;
+        case ROUGHSST_MODEL::LIMITER_AUPOIX: opts.limiter_aupoix = true; break;
```

Comment:
Can you add a check that mutually exclusive options are not used together please?
It's either 98 or 06 and one limiter or the other right?
(we do this for other options of this type too)

Reply:
Ah sorry I misunderstood that